src/transformers/hyperparameter_search.py

```diff
@@ -0,0 +1,86 @@
+from typing import Callable, Dict, Type
+
+from .integrations import (
+    is_optuna_available,
+    is_ray_available,
+    is_sigopt_available,
+    is_wandb_available,
+    run_hp_search_optuna,
+    run_hp_search_ray,
+    run_hp_search_sigopt,
+    run_hp_search_wandb,
+)
+from .trainer_utils import (
+    HPSearchBackend,
+    default_hp_space_optuna,
+    default_hp_space_ray,
+    default_hp_space_sigopt,
+    default_hp_space_wandb,
+)
+
+
+class HyperParamSearchBackendBase:
+    name: HPSearchBackend
```

Comment:
Let's leave the name as string here, but have the dictionary you build later be enum->backend class.